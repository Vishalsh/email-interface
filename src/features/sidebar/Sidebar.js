import React from 'react';
import { Link } from 'react-router-dom';

import routes from 'constants/routes';
import classes from './Sidebar.module.scss';

const links = [{
  name: 'Dashboard',
  icon: 'dashboard',
  routeTo: routes.DASHBOARD
}, {
  name: 'Layouts',
  icon: 'layouts',
  routeTo: routes.LAYOUTS
}, {
  name: 'Graphs',
  icon: 'graphs',
  routeTo: routes.GRAPHS
}, {
  name: 'Mailbox',
  icon: 'mailbox',
  routeTo: routes.MAILBOX
}, {
  name: 'Metrics',
  icon: 'metrics',
  routeTo: routes.METRICS
}, {
  name: 'Widgets',
  icon: 'widgets',
  routeTo: routes.WIDGETS
}, {
  name: 'Forms',
  icon: 'forms',
  routeTo: routes.FORMS
}, {
  name: 'App Views',
  icon: 'app-views',
  routeTo: routes.APP_VIEWS
}];

const Sidebar = () => {
  return (
    <nav className={classes.nav}>
      {
        links.map((link, index) => (
          <li key={`${link.name}_${index}`}>
            <Link className={classes.link} to={link.routeTo}>
              <i className={`icon icon-${link.icon} ${classes.linkIcon}`} />
            </Link>
          </li>
        ))
      }
    </nav>
  )
};

export default Sidebar;
