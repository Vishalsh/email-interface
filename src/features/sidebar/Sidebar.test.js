import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Sidebar from './Sidebar';

describe('Sidebar', () => {
  it('should render the component', () => {
    expect(toJson(shallow(<Sidebar/>))).toMatchSnapshot();
  })
});