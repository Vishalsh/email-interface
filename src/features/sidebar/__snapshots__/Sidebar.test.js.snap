// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar should render the component 1`] = `
<nav
  className="nav"
>
  <li
    key="Dashboard_0"
  >
    <Link
      className="link"
      replace={false}
      to="/dashboard"
    >
      <i
        className="icon icon-dashboard linkIcon"
      />
    </Link>
  </li>
  <li
    key="Layouts_1"
  >
    <Link
      className="link"
      replace={false}
      to="/layouts"
    >
      <i
        className="icon icon-layouts linkIcon"
      />
    </Link>
  </li>
  <li
    key="Graphs_2"
  >
    <Link
      className="link"
      replace={false}
      to="/graphs"
    >
      <i
        className="icon icon-graphs linkIcon"
      />
    </Link>
  </li>
  <li
    key="Mailbox_3"
  >
    <Link
      className="link"
      replace={false}
      to="/mailbox"
    >
      <i
        className="icon icon-mailbox linkIcon"
      />
    </Link>
  </li>
  <li
    key="Metrics_4"
  >
    <Link
      className="link"
      replace={false}
      to="/metrics"
    >
      <i
        className="icon icon-metrics linkIcon"
      />
    </Link>
  </li>
  <li
    key="Widgets_5"
  >
    <Link
      className="link"
      replace={false}
      to="/widgets"
    >
      <i
        className="icon icon-widgets linkIcon"
      />
    </Link>
  </li>
  <li
    key="Forms_6"
  >
    <Link
      className="link"
      replace={false}
      to="/forms"
    >
      <i
        className="icon icon-forms linkIcon"
      />
    </Link>
  </li>
  <li
    key="App Views_7"
  >
    <Link
      className="link"
      replace={false}
      to="/app-views"
    >
      <i
        className="icon icon-app-views linkIcon"
      />
    </Link>
  </li>
</nav>
`;
